File: 121-best-time-to-buy-and-sell-stock/js/brute-force.js
/**
 * @param {number[]} prices
 * @return {number}
 */
const maxProfit = function(prices) {
  let maxProfit = 0
  for (let i = 0; i < prices.length; i++){
      let localMax = 0
      for(let j = i + 1; j < prices.length; j ++){
          localMax = Math.max(localMax, prices[j]) 
      }
      maxProfit = Math.max(maxProfit, localMax - prices[i])
  }
  return maxProfit
  
};

File: 121-best-time-to-buy-and-sell-stock/python/brute-force.py
'''
    STATUS: Time Limit Exceeded
    Time Complexity: 0(n^2)
    Space Complexity: 0(1)
'''
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        global_max = 0

        for i in range(len(prices)):
            local_max = prices[i]
            for j in range(i + 1, len(prices)):
                local_max = max(local_max, prices[j])
            global_max = max(global_max, local_max - prices[i])
        return global_max
